cmd/cel-shed/p2p.go

```diff
@@ -75,3 +85,124 @@ var p2pPeerIDCmd = &cobra.Command{
 	},
 	Args: cobra.ExactArgs(1),
 }
+
+var (
+	errorOnAnyFailure bool
+	errorOnAllFailure bool
+	connectionTimeout time.Duration
+)
+
+var p2pConnectBootstrappersCmd = &cobra.Command{
+	Use:   "connect-bootstrappers [network]",
+	Short: "Connect to bootstrappers of a certain network",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if errorOnAnyFailure && errorOnAllFailure {
+			return fmt.Errorf("only one of --err-any and --err-all can be specified")
+		}
+
+		ctx, cancel := context.WithTimeout(cmd.Context(), connectionTimeout)
+		defer cancel()
+
+		network := p2p.GetNetwork(args[0])
+		bootstrappers, err := p2p.BootstrappersFor(network)
+		if err != nil {
+			return fmt.Errorf("failed to get bootstrappers: %w", err)
+		}
+
+		store := nodebuilder.NewMemStore()
+		cfg := p2p.DefaultConfig(node.Light)
+		modp2p := p2p.ConstructModule(node.Light, &cfg)
+
+		var mod p2p.Module
+		app := fx.New(
+			fx.NopLogger,
+			modp2p,
+			fx.Provide(fraud.Unmarshaler),
+			fx.Provide(cmd.Context),
+			fx.Provide(store.Keystore),
+			fx.Provide(store.Datastore),
+			fx.Supply(bootstrappers),
+			fx.Supply(network),
+			fx.Supply(node.Light),
+			fx.Invoke(func(modprov p2p.Module) {
+				mod = modprov
+			}),
+		)
+
+		if err := app.Start(ctx); err != nil {
+			return fmt.Errorf("failed to start app: %w", err)
+		}
+		defer func() {
+			if err := app.Stop(ctx); err != nil {
+				fmt.Printf("failed to stop application: %v\n", err)
+			}
+		}()
+
+		p2pInfo, err := mod.Info(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get p2p info: %w", err)
+		}
+
+		fmt.Printf("PeerID: %s\n", p2pInfo.ID)
+		for _, addr := range p2pInfo.Addrs {
+			fmt.Printf("Listening on: %s\n", addr.String())
+		}
+		fmt.Println()
+
+		successfulConnections := 0
+		failedConnections := 0
```

Comment:
This one should be `var failedConnections atomic.Int32` 'cause it's accessed in different goroutines. Same for `successfulConnections`.

Reply:
Ah...you're using a mutex for this, my bad 🙈 

Anyway! you can replace type to `atomic.Int32` and drop the mutex.